src/Tools/GenerateDocumentationAndConfigFiles/Program.cs

```diff
@@ -1435,8 +1464,10 @@ private enum RulesetKind
         private enum AnalysisMode
         {
             Default,
-            AllDisabledByDefault,
-            AllEnabledByDefault,
+            None,
+            Minimum,
+            Recommended,
+            All
```

Comment:
Isn't this a breaking change?

Reply:
No, the updates to the generated targets above ensure back compat and that old values are correctly mapped: `AllEnabledByDefault` will be mapped to `All` and `AllDisabledByDefault` will be mapped to `None`.

Reply:
I have also manually verified this by trying out the locally built package.

Reply:
@mavasani The comment above the enum is now misleading I think.

It assumes that the names in the enum are used directly in AnalysisLevel MSBuild property.